### PR TITLE
Refactor holders which decorate another holder

### PR DESCRIPTION
**Pull request by WonderCsabo**
This PR contains one commit which was split off from #1294.

### CODE REVIEW

File: AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/holder/ReceiverRegistrationHolder.java [hunk 1]
```diff
@@ -34,14 +34,13 @@
 import com.sun.codemodel.JExpression;
 import com.sun.codemodel.JFieldVar;
 
-public class ReceiverRegistrationHolder<T extends EComponentHolder & HasReceiverRegistration> {
+public class ReceiverRegistrationHolder extends GeneratedClassHolderDecorator<EComponentHolder> {
```
Comment (yDelouis) [inline AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/holder/ReceiverRegistrationHolder.java]: Why don't you keep `ReceiverRegistrationHolder` generic ? It prevent us from casting and clearly state that the `holder` must implement `HasReceiverRegistration`.

**Comment by yDelouis** [comment]
I'm okay with the purpose of this PR but I would have kept `ReceiverRegistrationHolder` generic (see my comment).

File: AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/holder/ReceiverRegistrationHolder.java (hunk 1 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/holder/ReceiverRegistrationHolder.java]: OK, i'll add back the type parameter.

**Comment by WonderCsabo** [comment]
I addressed your review comment.

**Pull request merged**